xayn-ai/src/coi/point.rs

```diff
@@ -183,78 +183,50 @@ impl From<UserInterests_v0_2_0> for UserInterests {
     }
 }
 
-/// Finds the closest centre of interest (CoI) for the given embedding.
-///
-/// Returns the index of the CoI along with the weighted distance between the given embedding
-/// and the k nearest CoIs. If no CoIs were given, `None` will be returned.
+/// Finds the most similar centre of interest (CoI) for the given embedding.
 pub(super) fn find_closest_coi_index(
     cois: &[impl CoiPoint],
     embedding: &Embedding,
-    neighbors: usize,
 ) -> Option<(usize, f32)> {
     if cois.is_empty() {
         return None;
     }
 
-    let mut distances = cois
+    let mut similarities = cois
         .iter()
-        .map(|coi| l2_distance(embedding.view(), coi.point().view()))
+        .map(|coi| cosine_similarity(embedding.view(), coi.point().view()))
         .enumerate()
         .collect::<Vec<_>>();
-    distances.sort_by(|(_, this), (_, other)| nan_safe_f32_cmp(this, other));
-    let index = distances[0].0;
-
-    let total = distances.iter().map(|&(_, distance)| distance).sum::<f32>();
-    let distance = (total > 0.)
-        .then(|| {
-            distances
-                .iter()
-                .take(neighbors)
-                .zip(distances.iter().take(neighbors).rev())
-                .map(|((_, distance), (_, reversed))| distance * (reversed / total))
-                .sum()
-        })
-        .unwrap_or_default();
-
-    Some((index, distance))
+
+    similarities.sort_by(|(_, this), (_, other)| other.partial_cmp(this).unwrap());
```

Comment:
i'd prefer to use `reverse()` to make it obvious that the sorting order is reversered
```suggestion
    similarities.sort_by(|(_, this), (_, other)| this.partial_cmp(other).unwrap().reverse());
```